### PR TITLE
Support ktx2 loader

### PR DESCRIPTION
**Pull request by gz65555**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/galacean/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
Support ktx2 loader
- doc: https://yuque.antfin-inc.com/kevin.hs/notebook/im0sldctixwmx5gl?singleDoc# (not public now)
- demo: https://github.com/gz65555/ktx2-demo (link engine)

#### Basic
```typescript
engine.resourceManager('skybox.ktx2').then(...);
engine.resourceManager('duck.ktx2').then(...);
```

#### Init Transcode WebWorker
```typescript
KTX2Loader.init(...);
```

#### Destroy Transcode WebWorker
```typescript
KTX2Loader.destroy();
```

#### Specify Format Priority(Advanced)
```typescript
engine.resourceManager({
...,
formatPriorities: [KTX2TargetFormat.RGBA32]
}).then(...);
```

### CODE REVIEW

**Comment by GuoLei1990** [comment]
`BinomialLLCTranscoder` and `KhronosTranscoder` folder name should be lowercase

File: packages/loader/src/ktx2/WorkerPool.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
+// @ts-nocheck
+interface TaskItem<T, U> {
+  message: T;
+  transfer?: Array<Transferable | OffscreenCanvas>;
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/WorkerPool.ts:4]: `Transferable` already contain `OffscreenCanvas`

File: packages/loader/src/ktx2/WorkerPool.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
+// @ts-nocheck
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/WorkerPool.ts:1]: Why need `@ts-nocheck`?

File: packages/loader/src/ktx2/WorkerPool.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  private _workers: Array<Worker>;
+  private _preparedResolve: any[] = [];
+
+  /**
+   * Constructor of WorkerPool.
+   * @param limitedCount - worker limit count
+   * @param _workerCreator - creator of worker
+   */
+  constructor(public readonly limitedCount = 4, private readonly _workerCreator: () => Worker) {
+    this._workers = new Array<Worker>(limitedCount);
```
Comment (gz65555) [inline packages/loader/src/ktx2/WorkerPool.ts:26]: _workerCreator 修改成异步初始化

File: packages/loader/src/index.ts [hunk 4]
```diff
@@ -14,11 +15,13 @@ import "./SpriteAtlasLoader";
...
+import "./ktx2/KTX2Loader";
 
-export { parseSingleKTX } from "./compressed-texture";
 export type { GLTFParams } from "./GLTFLoader";
-export * from "./resource-deserialize";
 export * from "./SceneLoader";
 export type { Texture2DParams } from "./Texture2DLoader";
+export { parseSingleKTX } from "./compressed-texture";
 export * from "./gltf";
+export { KTX2Loader } from "./ktx2/KTX2Loader";
```
Comment (GuoLei1990) [inline packages/loader/src/index.ts:25]: should clear

File: packages/loader/src/index.ts [hunk 5]
```diff
@@ -14,11 +15,13 @@ import "./SpriteAtlasLoader";
...
+export { KTX2TargetFormat } from "./ktx2/KTX2TargetFormat";
```
Comment (GuoLei1990) [inline packages/loader/src/index.ts:26]: Should export to user like `Texture2DParams`

File: packages/loader/src/ktx2/KTX2TargetFormat.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+export enum KTX2TargetFormat {
+  ASTC,
+  BC7,
+  DXT,
+  PVRTC,
+  ETC,
+  R8,
+  RG8,
+  RGBA32
+}
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/KTX2TargetFormat.ts:13]: - Add simple comments 

- Use `RGBA8`,  I will opt engine format.

File: packages/loader/src/ktx2/KTX2TargetFormat.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+/**
+ * KTX2 transcode format.
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/KTX2TargetFormat.ts:2]: KTX2 transcode target format.

File: packages/loader/src/ktx2/WorkerPool.ts [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+  reject: (reason?: any) => void;
+}
+
+/**
+ * @internal
+ * WorkerPool, T is return type, U is post message type.
+ */
+export class WorkerPool<T = any, U = any> {
+  private _taskQueue: TaskItem<T, U>[] = [];
+  private _resolveMap: Array<{ resolve: (item: U | PromiseLike<U>) => void; reject: (reason?: any) => void }> = [];
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/WorkerPool.ts:15]: Combine with `_workers` property

File: packages/loader/src/ktx2/WorkerPool.ts [hunk 9]
```diff
@@ -0,0 +1,106 @@
+// @ts-nocheck
+interface TaskItem<T, U> {
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/WorkerPool.ts:2]: Put this behind `WorkerPool` class

File: packages/loader/src/ktx2/BinomialLLCTranscoder/BinomialLLCTranscoder.ts [hunk 10]
```diff
@@ -0,0 +1,64 @@
+import { KTX2TargetFormat } from "../KTX2TargetFormat";
+import { TranscodeResult } from "../TranscodeResult";
+import { WorkerPool } from "../WorkerPool";
+import { IInitMessage, TranscodeWorkerCode } from "./TranscodeWorkerCode";
+
+/** @internal */
+export class BinomialLLCTranscoder {
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/BinomialLLCTranscoder/BinomialLLCTranscoder.ts:7]: Implement interface

File: packages/loader/src/ktx2/KTX2Loader.ts [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+    }
+    return null;
+  }
+
+  private static _getBinomialLLCTranscoder(workerCount: number = 4) {
+    if (!this._binomialLLCTranscoder) {
+      this._binomialLLCTranscoder = new BinomialLLCTranscoder(workerCount);
+    }
+    return this._binomialLLCTranscoder;
+  }
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/KTX2Loader.ts:112]: use 
```ts
private static _getBinomialLLCTranscoder(workerCount: number = 4) {
    return (this._binomialLLCTranscoder ||= new BinomialLLCTranscoder(workerCount));
  }
```

File: packages/loader/src/ktx2/KTX2Loader.ts [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+@resourceLoader(AssetType.KTX, ["ktx2"])
+export class KTX2Loader extends Loader<Texture2D | TextureCube> {
+  private static _binomialLLCTranscoder: BinomialLLCTranscoder;
+  private static _khronosTranscoder: KhronosTranscoder;
+  private static _supportedMap = {
+    [KTX2TargetFormat.ASTC]: [GLCapabilityType.astc],
+    [KTX2TargetFormat.ETC]: [GLCapabilityType.etc],
+    [KTX2TargetFormat.DXT]: [GLCapabilityType.s3tc],
+    [KTX2TargetFormat.PVRTC]: [GLCapabilityType.pvrtc, GLCapabilityType.pvrtc_webkit]
+  };
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/KTX2Loader.ts:33]: `canIUse` should improve

File: packages/loader/src/ktx2/KTX2Loader.ts [hunk 13]
```diff
@@ -0,0 +1,192 @@
...
+   * @returns
+   */
+  static init(engine: WebGLEngine, workerCount: number = 4, formatPriorities?: KTX2TargetFormat[]): Promise<void> {
+    // @ts-ignore
+    if (this._decisionTargetFormat(engine._hardwareRenderer, formatPriorities) === KTX2TargetFormat.ASTC) {
+      return this._getKhronosTranscoder(workerCount).init();
+    } else {
+      return this._getBinomialLLCTranscoder(workerCount).init();
+    }
+  }
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/KTX2Loader.ts:49]: You `WebGLEngineConfiguration`

File: packages/loader/src/ktx2/KTX2Loader.ts [hunk 14]
```diff
@@ -0,0 +1,192 @@
...
+
+    if (targetFormat === null) {
+      Logger.warn("can't support any compressed texture, downgrade to RGBA8");
+      return KTX2TargetFormat.RGBA8;
+    }
+    // TODO support bc7: https://github.com/galacean/engine/issues/1371
+    return targetFormat;
+  }
+
+  private static _decisionTargetFormat(
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/KTX2Loader.ts:87]: _decisionSupportTargetFormat

File: packages/loader/src/ktx2/KTX2Loader.ts [hunk 15]
```diff
@@ -0,0 +1,192 @@
...
+   * Destroy ktx2 transcoder worker.
+   */
+  static destroy(): void {
+    if (this._binomialLLCTranscoder) this._binomialLLCTranscoder.destroy();
+    if (this._khronosTranscoder) this._khronosTranscoder.destroy();
+    this._binomialLLCTranscoder = null;
+    this._khronosTranscoder = null;
+  }
+
+  private static _detectTranscoder(
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/KTX2Loader.ts:61]: `_detectformat`

File: packages/loader/src/ktx2/KTX2Loader.ts [hunk 16]
```diff
@@ -0,0 +1,192 @@
...
+    }
+    if (basisFormat === KTX2TargetFormat.PVRTC && MathUtil.isPowerOf2(width) && MathUtil.isPowerOf2(height)) {
+      if (hasAlpha) return TextureFormat.PVRTC_RGBA4;
+      else return TextureFormat.PVRTC_RGB4;
+    }
+    if (basisFormat === KTX2TargetFormat.RGBA8) {
+      return TextureFormat.R8G8B8A8;
+    }
+  }
+}
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/KTX2Loader.ts:192]: Use `switch case` and `: ?`

File: packages/loader/src/ktx2/KTX2Loader.ts [hunk 17]
```diff
@@ -0,0 +1,192 @@
...
+      else return TextureFormat.ETC2_RGB;
+    }
+    if (basisFormat === KTX2TargetFormat.DXT) {
+      if (hasAlpha) return TextureFormat.DXT5;
+      else return TextureFormat.DXT1;
+    }
+    if (basisFormat === KTX2TargetFormat.PVRTC && MathUtil.isPowerOf2(width) && MathUtil.isPowerOf2(height)) {
+      if (hasAlpha) return TextureFormat.PVRTC_RGBA4;
+      else return TextureFormat.PVRTC_RGB4;
+    }
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/KTX2Loader.ts:187]: PVRTC width and height should same

File: packages/loader/src/ktx2/KTX2Loader.ts [hunk 18]
```diff
@@ -0,0 +1,192 @@
...
+  /**
+   * @internal
+   */
+  load(item: LoadItem, resourceManager: ResourceManager): AssetPromise<any> {
+    return this.request<ArrayBuffer>(item.url!, { type: "arraybuffer" }).then((buffer) => {
+      const ktx2Container = new KTX2Container(buffer);
+      const formatPriorities = item.params?.formatPriorities;
+      const targetFormat = KTX2Loader._detectTranscoder(resourceManager.engine, ktx2Container, formatPriorities);
+      let transcodeResultPromise: Promise<any>;
+      if (targetFormat === KTX2TargetFormat.ASTC && ktx2Container.isUASTC) {
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/KTX2Loader.ts:130]: `R8` `RG8` `RGBA8` should use `KhronosTranscoder` too!

File: packages/loader/src/ktx2/KTX2Loader.ts [hunk 19]
```diff
@@ -0,0 +1,192 @@
...
+        const mipmaps = faces[0];
+        const mipmap = mipmaps.length > 1;
+        const engineFormat = this._getEngineTextureFormat(targetFormat, result);
+        if (faceCount !== 6) {
+          const texture = new Texture2D(resourceManager.engine, width, height, engineFormat, mipmap);
+          for (let mipLevel = 0; mipLevel < mipmaps.length; mipLevel++) {
+            const { width, height, data } = mipmaps[mipLevel];
+            texture.setPixelBuffer(data, mipLevel, 0, 0, width, height);
+          }
+          const params = ktx2Container.keyValue["GalaceanTextureParams"] as Uint8Array;
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/KTX2Loader.ts:149]: TextureCube also need

File: packages/loader/src/ktx2/KTX2Loader.ts [hunk 20]
```diff
@@ -0,0 +1,192 @@
...
+        const { width, height, faces } = result;
+        const faceCount = faces.length;
+        const mipmaps = faces[0];
+        const mipmap = mipmaps.length > 1;
+        const engineFormat = this._getEngineTextureFormat(targetFormat, result);
+        if (faceCount !== 6) {
+          const texture = new Texture2D(resourceManager.engine, width, height, engineFormat, mipmap);
+          for (let mipLevel = 0; mipLevel < mipmaps.length; mipLevel++) {
+            const { width, height, data } = mipmaps[mipLevel];
+            texture.setPixelBuffer(data, mipLevel, 0, 0, width, height);
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/KTX2Loader.ts:147]: `texture.setPixelBuffer(data, mipLevel);` is OK

File: packages/loader/src/resource-deserialize/utils/BufferReader.ts [hunk 21]
```diff
@@ -9,6 +9,7 @@ export class BufferReader {
   private _dataView: DataView;
   private _littleEndian: boolean;
   private _offset: number;
+  private _baseOffset: number;
 
   public static imageMapping = {
```
Comment (GuoLei1990) [inline packages/loader/src/resource-deserialize/utils/BufferReader.ts:14]: not used anywhere

File: packages/loader/src/resource-deserialize/utils/BufferReader.ts [hunk 22]
```diff
@@ -22,6 +23,11 @@ export class BufferReader {
     this._dataView = new DataView(buffer);
     this._littleEndian = littleEndian;
     this._offset = byteOffset;
+    this._baseOffset = byteOffset;
+  }
+
+  get relativeOffset() {
```
Comment (GuoLei1990) [inline packages/loader/src/resource-deserialize/utils/BufferReader.ts:29]: maybe`position` is better

File: packages/loader/src/ktx2/KTX2Loader.ts (hunk 15 above)
Comment (gz65555) [reply packages/loader/src/ktx2/KTX2Loader.ts]: done

File: packages/loader/src/ktx2/KTX2Loader.ts (hunk 14 above)
Comment (gz65555) [reply packages/loader/src/ktx2/KTX2Loader.ts]: done

File: packages/loader/src/resource-deserialize/utils/BufferReader.ts
Comment (GuoLei1990) [inline packages/loader/src/resource-deserialize/utils/BufferReader.ts:1]: `class ImageData` can delete

File: packages/loader/src/ktx2/KTX2Loader.ts (hunk 17 above)
Comment (gz65555) [reply packages/loader/src/ktx2/KTX2Loader.ts:187]: done

File: packages/loader/src/ktx2/KTX2Loader.ts (hunk 19 above)
Comment (gz65555) [reply packages/loader/src/ktx2/KTX2Loader.ts:149]: done

File: packages/loader/src/ktx2/KTX2Loader.ts (hunk 20 above)
Comment (gz65555) [reply packages/loader/src/ktx2/KTX2Loader.ts:147]: done

File: packages/loader/src/resource-deserialize/utils/BufferReader.ts
Comment (gz65555) [reply packages/loader/src/resource-deserialize/utils/BufferReader.ts:1]: done

File: packages/loader/src/ktx2/KTX2Loader.ts [hunk 23]
```diff
@@ -0,0 +1,192 @@
...
+    if (
+      targetFormat === KTX2TargetFormat.PVRTC &&
+      (!MathUtil.isPowerOf2(ktx2Container.pixelWidth) || !MathUtil.isPowerOf2(ktx2Container.pixelHeight))
+    ) {
+      Logger.warn("pvrtc image need power of 2, downgrade to RGBA8");
+      return KTX2TargetFormat.RGBA8;
+    }
+
+    if (targetFormat === null) {
+      Logger.warn("can't support any compressed texture, downgrade to RGBA8");
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/KTX2Loader.ts:83]: 注释首字母大写

Comment (gz65555) [reply packages/loader/src/ktx2/KTX2Loader.ts]: done

File: packages/loader/src/ktx2/transcoder/BinomialLLCTranscoder.ts [hunk 24]
```diff
@@ -0,0 +1,64 @@
...
+
+/** @internal */
+export class BinomialLLCTranscoder {
+  public static MessageId = 0;
+
+  private _transcodeWorkerPool: WorkerPool;
+
+  private _initPromise: Promise<any>;
+
+  constructor(public readonly workerLimit: number) {}
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/transcoder/BinomialLLCTranscoder.ts:14]: workerLimitCount

File: packages/loader/src/ktx2/constants.ts [hunk 25]
```diff
@@ -0,0 +1,8 @@
+export enum SupercompressionScheme {
+  None,
+  BasisLZ,
+  Zstd,
+  ZLib
+}
+
+export const KHR_DF_SAMPLE_DATATYPE_SIGNED = 0x40;
```
[the remark this answers is not in the document]
Comment (gz65555) [reply packages/loader/src/ktx2/constants.ts]: done

File: packages/loader/src/ktx2/KTX2TargetFormat.ts (hunk 6 above)
Comment (gz65555) [reply packages/loader/src/ktx2/KTX2TargetFormat.ts:21]: done

File: packages/loader/src/ktx2/KTX2TargetFormat.ts [hunk 26]
```diff
@@ -0,0 +1,21 @@
...
+export enum KTX2TargetFormat {
+  /** RGB(A) compressed format, 128 bits per 4x4 pixel block. */
+  ASTC,
+  /** Unsupported format. */
+  BC7,
+  /** RGB(A) compressed format。*/
+  DXT,
+  /** RGB(A) compressed format。*/
+  PVRTC,
+  /** RGB(A) compressed format。*/
```
Comment (GuoLei1990) [inline packages/loader/src/ktx2/KTX2TargetFormat.ts:13]: 。is chinese

File: packages/loader/src/resource-deserialize/utils/BufferReader.ts (hunk 21 above)
Comment (gz65555) [reply packages/loader/src/resource-deserialize/utils/BufferReader.ts]: done

File: packages/loader/src/ktx2/WorkerPool.ts [hunk 27]
```diff
@@ -0,0 +1,106 @@
...
+  message: T;
+  transfer?: Array<Transferable | OffscreenCanvas>;
+  resolve: (item: U | PromiseLike<U>) => void;
+  reject: (reason?: any) => void;
+}
+
+/**
+ * @internal
+ * WorkerPool, T is return type, U is post message type.
```
[the remark this answers is not in the document]
Comment (gz65555) [reply packages/loader/src/ktx2/WorkerPool.ts]: done

File: packages/loader/src/ktx2/WorkerPool.ts (hunk 2 above)
Comment (gz65555) [reply packages/loader/src/ktx2/WorkerPool.ts]: done

File: packages/loader/src/ktx2/WorkerPool.ts (hunk 3 above)
Comment (gz65555) [reply packages/loader/src/ktx2/WorkerPool.ts:24]: done

File: packages/loader/src/ktx2/WorkerPool.ts [hunk 28]
```diff
@@ -0,0 +1,106 @@
...
+      if (this._taskQueue.length) {
+        const { resolve, message, reject } = this._taskQueue.shift() as TaskItem<T, U>;
+        this._resolveMap[workerId] = { resolve, reject };
+        this._workers[workerId].postMessage(message);
+      } else {
+        this._workerStatus ^= 1 << workerId;
+      }
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (gz65555) [reply packages/loader/src/ktx2/WorkerPool.ts:104]: done

File: packages/loader/src/ktx2/WorkerPool.ts [hunk 29]
```diff
@@ -0,0 +1,106 @@
...
+   * @param limitedCount - worker limit count
+   * @param _workerCreator - creator of worker
+   */
+  constructor(public readonly limitedCount = 4, private readonly _workerCreator: () => Worker) {
+    this._workers = new Array<Worker>(limitedCount);
+  }
+
+  prepareWorker() {
+    const count = this.limitedCount;
+    const promises = new Array(count);
```
[the remark this answers is not in the document]
Comment (gz65555) [reply packages/loader/src/ktx2/WorkerPool.ts]: done

File: packages/loader/src/ktx2/BinomialLLCTranscoder/BinomialLLCTranscoder.ts (hunk 10 above)
Comment (gz65555) [reply packages/loader/src/ktx2/BinomialLLCTranscoder/BinomialLLCTranscoder.ts]: done

File: packages/loader/src/index.ts (hunk 5 above)
Comment (gz65555) [reply packages/loader/src/index.ts:26]: done

File: packages/loader/src/ktx2/KTX2TargetFormat.ts (hunk 7 above)
Comment (gz65555) [reply packages/loader/src/ktx2/KTX2TargetFormat.ts:2]: done

File: packages/loader/src/ktx2/WorkerPool.ts [hunk 30]
```diff
@@ -0,0 +1,106 @@
...
+    return Promise.all(promises);
+  }
+
+  /**
+   * Post message to worker.
+   * @param message - message which posted to worker
+   * @param transfer - message transfer
+   * @returns return a promise of message
+   */
+  postMessage(message: T): Promise<U> {
```
[the remark this answers is not in the document]
Comment (gz65555) [reply packages/loader/src/ktx2/WorkerPool.ts:42]: done

File: packages/loader/src/ktx2/WorkerPool.ts (hunk 9 above)
Comment (gz65555) [reply packages/loader/src/ktx2/WorkerPool.ts:1]: done

File: packages/loader/src/ktx2/KTX2Loader.ts (hunk 11 above)
Comment (gz65555) [reply packages/loader/src/ktx2/KTX2Loader.ts:115]: done

File: packages/loader/src/ktx2/WorkerPool.ts (hunk 9 above)
Comment (gz65555) [reply packages/loader/src/ktx2/WorkerPool.ts:1]: done

File: packages/loader/src/ktx2/WorkerPool.ts (hunk 8 above)
Comment (gz65555) [reply packages/loader/src/ktx2/WorkerPool.ts:14]: done

File: packages/loader/src/ktx2/transcoder/BinomialLLCTranscoder.ts (hunk 24 above)
Comment (gz65555) [reply packages/loader/src/ktx2/transcoder/BinomialLLCTranscoder.ts]: done

File: packages/loader/src/ktx2/KTX2Loader.ts (hunk 13 above)
Comment (gz65555) [reply packages/loader/src/ktx2/KTX2Loader.ts]: done

File: packages/loader/src/ktx2/TranscodeResult.ts [hunk 31]
```diff
@@ -0,0 +1,6 @@
+export interface TranscodeResult {
+  width: number;
+  height: number;
+  hasAlpha: boolean;
+  mipmaps: Array<{ data: Uint8Array; width: number; height: number }>;
+}
```
[the remark this answers is not in the document]
Comment (gz65555) [reply packages/loader/src/ktx2/TranscodeResult.ts]: done

File: packages/loader/src/ktx2/KhronosTranscoder/TranscoderWorkerCode.ts [hunk 32]
```diff
@@ -0,0 +1,229 @@
...
+type MessageType = "init" | "transcode";
+
+export interface IBaseMessage {
+  type: MessageType;
+}
+
+export interface IInitMessage extends IBaseMessage {
+  type: "init";
+  transcoderWasm: ArrayBuffer;
+}
```
[the remark this answers is not in the document]
Comment (gz65555) [reply packages/loader/src/ktx2/KhronosTranscoder/TranscoderWorkerCode.ts]: done

File: packages/loader/src/resource-deserialize/utils/BufferReader.ts [hunk 33]
```diff
@@ -22,6 +23,11 @@ export class BufferReader {
...
+    return this._offset - this._baseOffset;
   }
 
   get offset() {
```
[the remark this answers is not in the document]
Comment (gz65555) [reply packages/loader/src/resource-deserialize/utils/BufferReader.ts:20]: done

**Review by GuoLei1990 — APPROVED** [review]

**Pull request merged**